tests/files/tests.py

```diff
@@ -38,6 +38,20 @@ def test_context_manager(self):
         self.assertTrue(f.closed)
         self.assertTrue(orig_file.closed)
 
+    def test_open_resets_opened_file_to_start_and_returns_context_manager(self):
+        file = File(BytesIO(b'content'))
+        file.read()
+        with file.open() as f:
+            self.assertEqual(f.read(), b'content')
+
+    def test_open_reopens_closed_file_and_returns_context_manager(self):
+        file = File(tempfile.NamedTemporaryFile(delete=False))
+        file.close()
+        with file.open() as f:
+            self.assertFalse(f.closed)
+        # remove temporary file
+        os.unlink(file.name)
```

Comment:
This should be written in a way that it still works if open etc above throws an exception